### ISSUE TITLE
the code about complementary masking and loss calculation in V2?

### ISSUE THREAD

**Issue by guoyd1999** [NVlabs/Fast-dLLM#30]
Hi

Thank you for sharing the training code of V2. However, I only find the padding code in finetuner.py and can not find the code about complementary masking and block-wise loss calculation. Could you tell me where is it?

**Comment by ZinYY** [comment]
Same question
